### PR TITLE
Yc 591

### PR DESCRIPTION
**Pull request by monodo**
@rbovard please review

### CODE REVIEW

File: permits/api.py [hunk 1]
```diff
@@ -59,6 +59,14 @@ def list(self, request):
 
 
 class PermitRequestGeoTimeViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    Events request endpoint Usage:
```
Comment (rbovard) [inline permits/api.py:63]:
```suggestion
    Events request endpoint usage:
```

File: permits/views.py [hunk 2]
```diff
@@ -255,7 +249,6 @@ def get_context_data(self, **kwargs):
                 "directives": services.get_permit_request_directives(
                     self.permit_request
                 ),
-                "prolongation_enabled": prolongation_enabled,
```
Comment (rbovard) [inline permits/views.py:258]: Uh? Rebase issue?

File: permits/views.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 from django.core.exceptions import PermissionDenied, SuspiciousOperation
 from django.core.files.base import ContentFile
 from django.db import transaction
-from django.db.models import Prefetch, Sum
```
Comment (rbovard) [inline permits/views.py:16]: Uh? Rebase issue?

File: permits/views.py [hunk 4]
```diff
@@ -305,8 +298,20 @@ def get_amend_form(self, data=None):
         ):
             # Only set the `status` default value if it's submitted for validation, to prevent accidentally resetting
             # the status
+
+            first_wot_type = (
+                services.get_works_object_type_choices(self.permit_request)
+                .first()
```
Comment (rbovard) [inline permits/views.py:304]: And if there is mutliple names?

File: permits/models.py [hunk 5]
```diff
@@ -427,7 +427,14 @@ class PermitRequest(models.Model):
         (PROLONGATION_STATUS_APPROVED, _("Approuvée")),
         (PROLONGATION_STATUS_REJECTED, _("Refusée")),
     )
-
+    shortname = models.CharField(
+        _("nom de la demande"),
+        max_length=32,
```
Comment (rbovard) [inline permits/models.py:432]: In the story it's 64

File: permits/models.py [hunk 6]
```diff
@@ -427,7 +427,14 @@ class PermitRequest(models.Model):
...
+        help_text=_(
+            "Sera affiché dans le calendrier si la demande est rendue tout publique, ex: Brandons, max. 32 caractères"
```
Comment (rbovard) [inline permits/models.py:434]: In the story it's 64

Comment (rbovard) [inline permits/models.py:434]:
```suggestion
            "Sera affiché dans le calendrier si la demande est rendue tout publique, ex: Brandons (max. 32 caractères)"
```

File: permits/forms.py [hunk 7]
```diff
@@ -828,7 +828,11 @@ class PermitRequestAdditionalInformationForm(forms.ModelForm):
 
     class Meta:
         model = models.PermitRequest
-        fields = ["is_public", "status"]
+        fields = [
+            "is_public",
+            "shortname",
```
Comment (rbovard) [inline permits/forms.py:833]: Hide this field if `ENABLE_GEOCALENDAR` is false (like is_public)

**Review by rbovard — CHANGES_REQUESTED** [review]

File: permits/views.py (hunk 2 above)
Comment (monodo) [reply permits/views.py:258]: Thanks, I'll start again from main, messed a rebased, indeed

File: permits/models.py (hunk 5 above)
Comment (monodo) [reply permits/models.py:432]: Toooo long after checking with @benschnell

File: permits/forms.py (hunk 7 above)
Comment (monodo) [reply permits/forms.py:833]: Sure ? A short name ight be helpful anyway IMHO

Comment (rbovard) [reply permits/forms.py:833]: > Sera affiché dans le calendrier si la demande est rendue tout publique

It doen't make sense if no calendar...

File: permits/views.py (hunk 4 above)
Comment (monodo) [reply permits/views.py:304]: It's juste a automated suggestion, that has to be accepted or edited by pilot

File: permits/forms.py (hunk 7 above)
Comment (monodo) [reply permits/forms.py:833]: Maybe in the list view or in the printed pdf  ?

**Comment by monodo** [comment]
replaced by https://github.com/yverdon/geocity/pull/401

**Pull request closed without merging**